packages/integration-sdk-runtime/src/logger/__tests__/index.test.ts

```diff
@@ -680,3 +667,68 @@ describe('#publishErrorEvent', () => {
     });
   });
 });
+
+describe('#handleFailure', () => {
+  test('should invoke callback on handleFailure', () => {
+    const logger = createIntegrationLogger({
+      name,
+      invocationConfig,
+    });
+
+    const errorSpy = jest.spyOn(logger, 'error');
+    const publishEventSpy = jest.spyOn(logger, 'publishEvent');
+
+    const onFailureSpy = jest.fn();
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    // Property 'onFailure' is private and only accessible within class 'IntegrationLogger'.ts(2341)
+    logger.onFailure = onFailureSpy;
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    // Property 'handleFailure' is private and only accessible within class 'IntegrationLogger'.ts(2341)
+    logger.handleFailure({
+      err: new Error(),
+      errorId: 'SOME_ERROR_ID',
+      eventName: 'SOME_EVENT',
+      description: 'an error :(',
+    });
+
+    expect(onFailureSpy).toHaveBeenCalledTimes(1);
+    expect(publishEventSpy).toHaveBeenCalledTimes(1);
+
+    // should call error
+    expect(errorSpy).toHaveBeenCalledTimes(1);
+  });
+
+  test('should invoke callback on handleFailure', () => {
```

Comment:
This is the same test description as the previous test.

Reply:
Meant to test the difference between an error that satisfies `shouldReportErrorToOperator`. Fixed.

Reply:
Looking at the expectations later in the functions, I see the differences to be essentially

```
if error for operator {
  (expect(errorSpy).toHaveBeenCalledTimes(1)
} else {
  expect(warnSpy).toHaveBeenCalledTimes(1);
}
```

Perhaps there could be three tests and descriptions could be:

* `#handleFailure should invoke onFailure`
* `#handleFailure should log as warn when error is meant to be handled by a user`
* `#handleFailure should log as error when error is meant to be handled by an operator`